jdl/converters/jdl-to-json/jdl-with-applications-to-json-converter.js

```diff
@@ -165,3 +166,13 @@ function formatEntitiesForEachApplication(entitiesForEachApplication) {
         entitiesForEachApplication.set(applicationName, Object.values(applicationEntities));
     });
 }
+
+function addApplicationsWithoutEntities(entitiesForEachApplication) {
+    const applicationsWithoutEntities = jdlObject
+        .getApplications()
+        .filter(jdlApplication => jdlApplication.getEntityNames().length === 0)
+        .map(jdlApplication => jdlApplication.getConfigurationOptionValue('baseName'));
+    if (applicationsWithoutEntities.length > 0) {
+        applicationsWithoutEntities.forEach(applicationName => entitiesForEachApplication.set(applicationName, []));
+    }
+}
```

Comment:
if this is the solution, we could do it in a more optimized way:
  - loop through the applications (`getApplications()` then forEach)
  - in the forEach loop, check if there are assigned entities and immediately add them

Reply:
Agreed, infact there is a `jdlObject.forEachApplication()` for iteration over applications.